lib/sufia/import/generic_file_translator.rb

```diff
@@ -0,0 +1,42 @@
+module Sufia::Import
+  # Imports a Sufia 6.0-exported GenericFile into a Sufia PCDM GernericWork and FileSet
+  class GenericFileTranslator
+    # @param Hash settings needed to pass through to builder
+    #        @attr sufia6_user      - User name for accessing the sufia 6 fedora
+    #        @attr sufia6_password  - Password for accessing the sufia 6 fedora
+    #        @attr import_binary    - (default true) Import the binary content from fedora
+    def initialize(settings)
+      # TODO: work builder doesn't need settings; just file builder
+      @work_builder = WorkBuilder.new(settings)
+      @file_set_builder = FileSetBuilder.new(settings)
+    end
+
+    # loop through the files found in import_dir and map each GenericFile to a Sufia.primary_work and FileSet
+    # @param import_dir a directory containing export files, e.g.: generic_file_[id].json
+    # @return list of imported files
+    def import(import_dir)
+      # TODO: error instead of return
+      return unless Dir.exist?(import_dir)
+      # turn import_dir into a list of actual filenames
+      files = Dir.glob(File.join(import_dir, '*')).select { |f| exported_gf?(f) }
+      files.each { |f| import_file(f) }
+      files
+    end
+
+    private
+
+      def import_file(filename)
+        file_str = File.read(filename)
+        json = JSON.parse(file_str)
+        work = @work_builder.build(json)
+        fileset = @file_set_builder.build(json)
+        work.ordered_members << fileset
+      end
+
+      def exported_gf?(f)
+        File.file?(f) &&
+          File.extname(f) == '.json' &&
+          File.basename(f) =~ /^generic_file.*/
```

Comment:
We may want this to be customizable.  Not sure it is needed, since that is exactly the string that is created by https://github.com/projecthydra/sufia/blob/6.x-stable/lib/sufia/export/actor.rb#L81

I like the validation!